### PR TITLE
Factor out guards into helpers in balanceTransaction

### PR DESCRIPTION
**Pull request by Anviking**
- [x] Factor out guards in balanceTransaction into helpers for readability

### Comments

Preparation for other work.

### Issue Number

ADP-1372

### CODE REVIEW

File: lib/core/src/Cardano/Wallet.hs [hunk 1]
```diff
@@ -1684,6 +1619,80 @@ balanceTransaction
...
+                Cardano.TxWithdrawals _ wdrls -> Set.size
+                    (Set.fromList $ map networkOfWdrl wdrls) > 1
+        when conflictingWdrlNetworks $
+            throwE $ ErrBalanceTxNotYetSupported ConflictingNetworks
+
+    guardExistingCollateral = do
+        -- Coin selection does not support pre-defining collateral. In Sep 2021
+        -- consensus was that we /could/ allow for it with just a day's work or so,
+        -- but that the need for it was unclear enough that it was not in any way
+        -- a priority.
```
Comment (jonathanknowles) [inline lib/core/src/Cardano/Wallet.hs:1689]: I might be wrong, but suspect it would not just be "a day's work" 😄 .

**Review by jonathanknowles — APPROVED** [review]

File: lib/core/src/Cardano/Wallet.hs (hunk 1 above)
Comment (Anviking) [reply lib/core/src/Cardano/Wallet.hs:1689]: 😄 I thought that was what you said

**Comment by Anviking** [comment]
Thanks!
bors r+

**Comment by Anviking** [comment]
Bors r+

**Pull request merged**